### PR TITLE
[ feature/EWM1-167 ] : location api, itinerary model changes

### PR DESCRIPTION
**Pull request by jithinvv4**
### DESCRIPTION

**Location API Creation and Updates, Location Delete API, Itinerary Model Changes**

1. **Location API Creation and Updates**
   - Implemented APIs to create and update locations, including destinations and states. The API allows for multiselect of destinations when creating or updating state.

2. **Location Delete API**
   - Added a delete API for locations to allow for the removal of unnecessary or outdated location entries. This API provides administrators with the ability to manage location data more effectively by deleting locations as needed.

3. **Itinerary Model Changes**
   - Enhanced the itinerary model by adding the `important_message` and `things_to_carry` fields. These fields allow for the inclusion of important messages and lists of items to carry for each itinerary.

### REVIEWER
- @lenate-flycatch 

### CHECK LIST
- [x] I have performed a self-review of my own code.
- [x] My code follows the style guidelines of this project.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.

### CODE REVIEW

File: api/v1/general/serializers.py [hunk 1]
```diff
@@ -18,3 +18,21 @@ class CitySerializer(serializers.ModelSerializer):
...
+
+class LocationSerializer(serializers.ModelSerializer):
+    id = serializers.IntegerField(required=False)
+    
+    class Meta:
+        model = Location
+        fields = ['id', 'country', 'state', 'destinations']
+
+
+# class LocationGetSerializer(serializers.ModelSerializer):
```
Comment (lenate-flycatch) [inline api/v1/general/serializers.py]: remove commented code

File: api/v1/general/viewsets.py [hunk 2]
```diff
@@ -20,4 +22,18 @@ class CityViewSet(viewsets.ModelViewSet):
...
\ No newline at end of file
+    filterset_class = CityFilter
+
+
+class LocationViewSet(viewsets.ModelViewSet):
+    queryset = Location.objects.all()
+    serializer_class = LocationSerializer
+    http_method_names = ['get', 'delete']
+
+    def destroy(self, request, *args, **kwargs):
```
Comment (lenate-flycatch) [inline api/v1/general/viewsets.py:33]: add try except

File: api/v1/package/viewsets.py [hunk 3]
```diff
@@ -118,20 +118,20 @@ def create(self, request, *args, **kwargs):
             'id': serializer.data['id'],
             'statusCode': status.HTTP_201_CREATED}, status=status.HTTP_201_CREATED)
 
-    def handle_exception(self, exc):
-        # Call the default exception handler first to get the standard error response
-        response = super().handle_exception(exc)
+    # def handle_exception(self, exc):
```
Comment (lenate-flycatch) [inline api/v1/package/viewsets.py:121]: ??

Comment (jithinvv4) [reply api/v1/package/viewsets.py:121]: need to verify function, will check and uncomment later.

**Pull request merged**